homeassistant/components/tplink/switch.py

```diff
@@ -38,9 +39,48 @@ async def async_setup_entry(
     else:
         entities.append(SmartPlugSwitch(device, coordinator))
 
+    entities.append(SmartPlugLedSwitch(device, coordinator))
+
     async_add_entities(entities)
 
 
+class SmartPlugLedSwitch(CoordinatedTPLinkEntity, SwitchEntity):
+    """Representation of switch for the LED of a TPLink Smart Plug."""
+
+    coordinator: TPLinkDataUpdateCoordinator
+
+    _attr_entity_category = ENTITY_CATEGORY_CONFIG
+
+    def __init__(
+        self, device: SmartDevice, coordinator: TPLinkDataUpdateCoordinator
+    ) -> None:
+        """Initialize the LED switch."""
+        super().__init__(device, coordinator)
+
+        self._attr_name = f"{device.alias} LED"
+        self._attr_unique_id = f"{self.device.mac}_led"
+
+    @property
+    def icon(self) -> str:
+        """Return the icon for the LED."""
+        return "mdi:led-on" if self.is_on else "mdi:led-off"
+
+    @async_refresh_after
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the LED switch on."""
+        await self.device.set_led(True)
+
+    @async_refresh_after
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the LED switch off."""
+        await self.device.set_led(False)
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if LED switch is on."""
+        return bool(self.device.led)
```

Comment:
```suggestion
        return self.device.led
```
No need to force casting here as the library already reports it as a bool?

Reply:
Added this as `mypy` actually complains because the `.led` method of the lib does not have any declaration of type:

```
homeassistant/components/tplink/switch.py:81: error: Returning Any from function declared to return "bool"  [no-any-return]
Found 1 error in 1 file (checked 1 source file)
```

Reply:
This is fine for now

`python-kasa` needs to have a `py.typed` file added to the `kasa` dir for this to work.

Example:
https://github.com/jstasiak/python-zeroconf/blob/master/zeroconf/py.typed

Reply:
Hmh, that property is actually typed for both smartstrip and smartplug but having that cast here is not a dealbreaker for me.

Reply:
Oh, that's completely new to me, which feels really weird considering how long I've been using type hints so far... Anyway, I created a PR to add that file that allows removing the cast in the future.